tools/release.sh

```diff
@@ -0,0 +1,121 @@
+
+gpgkey=$(gpg --list-secret-keys | grep '^sec' | awk -F'( +|/)' '{print $3}')
+keycount=$(echo $gpgkey | wc -w)
+
+if [ $keycount -eq 0 ]; then
+  echo 'Need at least one GPG key, please make one with `gpg --gen-key`'
+  echo 'You will also need to submit your key to a public keyserver, e.g.'
+  echo '  https://sks-keyservers.net/i/#submit'
+  exit 1
+elif [ $keycount -ne 1 ]; then
+  echo -e 'You have multiple GPG keys:\n'
+
+  gpg --list-secret-keys
+
+  while true; do
+    echo $gpgkey | awk '{ for(i = 1; i <= NF; i++) { print i ") " $i; } }'
+    echo -n 'Select a key: '
+    read keynum
+
+    if $(test "$keynum" -eq "$keynum" > /dev/null 2>&1); then
+      _gpgkey=$(echo $gpgkey | awk '{ print $'${keynum}'}')
+      keycount=$(echo $_gpgkey | wc -w)
+      if [ $keycount -eq 1 ]; then
+        echo ""
+        gpgkey=$_gpgkey
+        break
+      fi
+    fi
+  done
+fi
+
+gpgfing=$(gpg --fingerprint $gpgkey | grep 'Key fingerprint =' | awk -F' = ' '{print $2}' | tr -d ' ')
+
+if ! test "$(grep $gpgfing README.md)"; then
+  echo 'Error: this GPG key fingerprint is not listed in ./README.md'
+  exit 1
+fi
+
+echo "Using GPG key: $gpgkey"
+echo "  Fingerprint: $gpgfing"
+
+################################################################################
+## Look for releases to promote
+
+echo -e "\n# Checking for releases ..."
+
+promotable=$(ssh ${webuser}@${webhost} $promotablecmd)
+
+if [ "X${promotable}" == "X" ]; then
+  echo "No releases to promote!"
+  exit 0
+fi
+
+echo -e "Found the following releases / builds ready to promote:\n"
+echo "$promotable" | sed 's/^/ * /'
+echo ""
+
+versions=$(echo "$promotable" | cut -d: -f1)
+
+################################################################################
+## Promote releases
+
+for version in $versions; do
+  while true; do
+    files=$(echo "$promotable" | grep "^${version}" | sed 's/^'${version}': //')
+    echo -n "Promote ${version} files (${files})? [y/n] "
+    yorn=""
+    read yorn
+
+    if [ "X${yorn}" == "Xn" ]; then
+      break
+    fi
+
+    if [ "X${yorn}" != "Xy" ]; then
+      continue
+    fi
+
+    echo "Promoting ${version}..."
+
+    promoteout=$(ssh ${webuser}@${webhost} $promotecmd $version)
+    shapath=$(echo "${promoteout}" | grep '^SHASUMS: ' | sed 's/^SHASUMS: //')
+    echo "${promoteout}" | grep -v '^SHASUMS: '
+
+    if [ "X${shapath}" == "X" ]; then
+      echo "Error: No SHASUMS file returned by promotion!"
+      exit 1
+    fi
+
+    echo "Signing SHASUMS for ${version}..."
+
+    shafile=$(basename $shapath)
+    shadir=$(dirname $shapath)
+    tmpdir="/tmp/_iojs_release.$$"
+
+    mkdir -p $tmpdir
+
+    scp ${webuser}@${webhost}:${shapath} ${tmpdir}/${shafile}
+
+    gpg --default-key $gpgkey --clearsign ${tmpdir}/${shafile}
+    gpg --default-key $gpgkey --armor --export --output ${tmpdir}/${shafile}.gpg
+
+    scp ${tmpdir}/${shafile} ${tmpdir}/${shafile}.asc ${tmpdir}/${shafile}.gpg ${webuser}@${webhost}:${shadir}/
```

Comment:
There might be a point in separating the actual upload so the person doing the release can review it "one last time" before a potential destructive action.